addons/binding/org.openhab.binding.neato/ESH-INF/thing/vacuumcleaner.xml

```diff
@@ -0,0 +1,204 @@
+		<item-type>String</item-type>
+		<label>Current available services</label>
+		<description>List of services that are currently available for the vacuum cleaner.</description>
+		<state readOnly="true" pattern="%s" />
+	</channel-type>
+
+	<channel-type id="version">
+		<item-type>String</item-type>
+		<label>Version</label>
+		<description>Version of the vacuum cleaner.</description>
+		<state readOnly="true" pattern="%s" />
+	</channel-type>
+
+	<channel-type id="model-name">
+		<item-type>String</item-type>
+		<label>Model Name</label>
+		<description>Model Name of the vacuum cleaner.</description>
+		<state readOnly="true" pattern="%s" />
+	</channel-type>
+
+	<channel-type id="firmware">
+		<item-type>String</item-type>
+		<label>Firmware</label>
+		<description>Firmware version of the vacuum cleaner.</description>
+		<state readOnly="true" pattern="%s" />
+	</channel-type>
+
+	<channel-type id="action">
+		<item-type>String</item-type>
+		<label>Current Action</label>
+		<description>Current action of the vacuum cleaner.</description>
+		<state readOnly="true" pattern="%s" />
+	</channel-type>
+
+	<channel-type id="dock-has-been-seen">
+		<item-type>Switch</item-type>
+		<label>Dock has been seen</label>
+		<description>True or False value if the dock has been seen</description>
+		<state readOnly="true" pattern="%s" />
+	</channel-type>
+
+    <channel-type id="is-docked">
+        <item-type>Switch</item-type>
+        <label>Is docked</label>
+        <description>Is the vacuum cleaner in the docking station?</description>
+        <state readOnly="true" pattern="%s" />
+    </channel-type>
+    
+    <channel-type id="is-scheduled">
+        <item-type>Switch</item-type>
+        <label>Is scheduled enabled</label>
+        <description>True or False value if the vacuum cleaner is scheduled for cleaning.</description>
+        <state readOnly="true" pattern="%s" />
+    </channel-type>
+    
+    <channel-type id="is-charging">
+        <item-type>Switch</item-type>
+        <label>Is Charging</label>
+        <description>Is the vacuum cleaner currently charging?</description>
+        <state readOnly="true" pattern="%s" />
+    </channel-type>
+    
+    <channel-type id="available-commands">
+        <item-type>String</item-type>
+        <label>Available Commands</label>
+        <description>List of available commands.</description>
+        <state readOnly="true" pattern="%s" />
+    </channel-type>
+
+    <channel-type id="error">
+        <item-type>String</item-type>
+        <label>Error</label>
+        <description>Last error message in system.</description>
+        <state readOnly="true" pattern="%s" />
+    </channel-type>
+
+
+    <channel-type id="command">
+        <item-type>String</item-type>
+        <label>Send Command</label>
+        <description>Send Commands to Vacuum Cleaner. (clean, pause, resume, stop, dock)</description>
+        <state readOnly="false" pattern="%s" />
```

Comment:
You should add all possible commands (clean, pause, stop, resume, etc) as options for the state. See more details from http://www.eclipse.org/smarthome/documentation/development/bindings/thing-definition.html